tests/components/pi_hole/test_init.py

```diff
@@ -0,0 +1,30 @@
+"""Test pi_hole component."""
+
+import unittest
+
+from homeassistant import setup
+from homeassistant.components import pi_hole
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+
+from asynctest import CoroutineMock
+from hole import Hole
+from tests.common import get_test_home_assistant
+from unittest.mock import patch
+
+
+class TestComponentPiHole(unittest.TestCase):
```

Comment:
We should no longer add tests that inherit from `unittest.TestCase`. Instead add standalone functions. 

```python
async def async_test_setup_no_config(hass):
    """Test a successful setup with no configuration."""
    with patch.object(Hole, "get_data", new=CoroutineMock()):
        assert await setup.async_setup_component(hass, pi_hole.DOMAIN, {"pi_hole": config})
```

Reply:
@balloob I'm having a hard time parsing this from both your comment and what's available in the developer docs.  

pytest isn't registering the result of the test (or even the test itself?) when it's defined as async outside of `unittest.TestCase`. I'm not able to find an existing integration that appears to implement this pattern, can you point me in the right direction?

Reply:
Tests function names must start with `test_`, so the example above isn't completely correct. See my example below for a working test.